README.md

```diff
@@ -168,6 +169,7 @@ Used by graphql gateway to fetch schema based on current containers
             "service_id": 3,
             "version": "ke9j34fuuei",
             "name": "service_a",
+            "url": "http://localhost:6111",
```

Comment:
The problem I see here is that you assume that there will be only one instance (IP) for this service, which in reality is not the case in big systems (like Pipedrive). We have up to 100 containers for a single service sometimes. 

Managing this service discovery and healthchecks is the responsibility of such services as Hashicorp's consul or kubernetes, as it needs high availability (and thus has clusters, leader elections etc). That's why I don't think that schema-registry should include such functionality.

Maybe it could ask service discovery instances for these URLs to avoid storing them in DB?
But then it should be extensible to define any kind of service discovery

Reply:
In our environments we also have multiple service instances for each service, and kubernetes handles the discovery and health checking aspects. Still, we have one logical URL per service, and that is this URL that we store in the registry, exactly like what's done in this PR. The registry thus does not have to take on the responsibility for service management, simply storing the logical URL per service. It makes for a pretty elegant GraphQL gateway implementation, as the gateway then only need to interact with the registry as its source of configuration for both schemas and service URLs.

If the registry were to work with service discovery instances, then that would mean an integration with Consul and other services of that nature as you're suggesting, adding greater complexity to the registry compared to simply storing service URLs.

My 2 cents :)

Reply:
I would like to +1 the need for the `url` field.

As it has been pointed out, the `url` field should serve as a logical locator for the service cluster (as opposed to a direct URL of a single service instance).

For example, it could be `http://some-service.consul` for people using Consul DNS, or it could be `http://some-service.svc` for people using HAProxy or some similar solution, or it could be `https://api.some-external-domain.com`, etc.

The point is that it allows one to use a mixture of service discovery mechanisms, which is valuable.

Reply:
ok, for own DNS urls it makes sense